src/dataviews/dataviews-collection.js

```diff
@@ -7,14 +7,26 @@ module.exports = Backbone.Collection.extend({
   initialize: function () {
     // If a histogram model applies the histogram sizes, rest should remove/disable
     // the sizes applied before.
-    this.bind('change:histogram_sizes', function (m, isSizesApplied) {
-      if (isSizesApplied) {
+    this.bind('change:autoStyle', function (m, autoStyle) {
+      if (autoStyle) {
         this.each(function (mdl) {
-          if (mdl !== m && mdl.get('histogram_sizes')) {
-            mdl.set('histogram_sizes', false);
+          if (mdl !== m && mdl.layer.get('layer_name') === m.layer.get('layer_name') && mdl.get('autoStyle')) {
```

Comment:
I don't see why cartodb.js needs to know if there is some autostyling, that is managed at different level, cartodb.js only knows (or should know) about cartocss for each layer